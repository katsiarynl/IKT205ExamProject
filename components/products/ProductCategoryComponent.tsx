import React from "react";
import { Pressable, View } from "react-native";
import { Card, Text } from "react-native-paper";
import products from "../../styles/products";
import { ProductCategoryComponentType } from "../../types/product";
// !100% the root of the problem
export default function ProductCategoryComponent(
  props: ProductCategoryComponentType
) {
  return (
    <View>
      <Pressable
        style={{ padding: 10 }}
        onPress={() => {
          props.category == "all"
            ? props.setCategory(["all"])
            : props.setCategory([props.category]);
        }}
      >
        <Card style={products.categoryitem}>
          <Text style={{ fontWeight: "bold" }}>{props["category"]}</Text>
        </Card>
      </Pressable>
    </View>
  );
}
