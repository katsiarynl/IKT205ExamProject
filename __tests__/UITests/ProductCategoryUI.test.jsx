import React from "react";
import { create } from "react-test-renderer";
import ProductCategoryComponent from "../../components/products/ProductCategoryComponent";
const tree = create(<ProductCategoryComponent category="Category" />);

jest.mock("react-native-paper", () => {
  return { Card: jest.fn() };
});

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
