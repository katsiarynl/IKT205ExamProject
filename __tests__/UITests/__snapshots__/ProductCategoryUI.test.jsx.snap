// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for ProductCategoryComponent 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "padding": 10,
      }
    }
  />
</View>
`;
